### PR TITLE
Feature greatuk 1232 wagtail caching middleware

### PR DESCRIPTION
**Pull request by davidu1975**
## What
We are adding wagtail caching middleware and nopage_cache and cache_page decorators
## Why
To improve Great CMS performance response times

_Tick or delete as appropriate:_

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GREATUK-1232
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Housekeeping

- [x] Added all new environment variables to Vault.
- [x] Python requirements have been re-compiled.
- [x] Frontend assets have been re-compiled.
- [x] I have checked that my PR is using the latest package versions of: great-components, directory-constants, directory-healthcheck, directory-validators, directory-components, directory-api-client, directory-ch-client, django-staff-sso-client, directory-forms-api-client, directory-sso-api-client, sigauth

### Performance
- [x] Evaluated the performance impact of the changes
- [x] Ensured that changes do not negatively affect application scalability.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: config/settings.py [hunk 1]
```diff
@@ -191,9 +193,18 @@
 else:
     REDIS_URL = env.str('REDIS_URL')
 
+# wagtail caching options
+# (see https://docs.coderedcorp.com/wagtail-cache/getting_started/django_settings.html#django-settings)
+WAGTAIL_CACHE = env.bool('WAGTAIL_CACHE', False)  # set to false for local
+WAGTAIL_CACHE_BACKEND = 'wagtail_cache'
```
Comment (hal274) [inline config/settings.py:199]: Can this cache key be called ``great_wagtail_cache`` or something better that indicates great's non-default key?

File: config/settings.py [hunk 2]
```diff
@@ -191,9 +193,18 @@
...
+WAGTAIL_CACHE_HEADER = True
```
Comment (hal274) [inline config/settings.py:200]: Does ``WAGTAIL_CACHE_HEADER = True``this default to false? If no what does setting it explicitly gives us?

File: config/settings.py (hunk 1 above)
Comment (davidu1975) [reply config/settings.py:199]: sure

File: config/settings.py (hunk 2 above)
Comment (davidu1975) [reply config/settings.py:200]: "By default, an HTTP header named X-Wagtail-Cache is added to the response to indicate a cache hit or miss. To turn off this header, set WAGTAIL_CACHE_HEADER = False, or to customize the header set to a string."
yes True is the default - but like to explicitly define it - I can set it to False or remove it

Comment (hal274) [reply config/settings.py:200]: Let's leave it. Frameworks can sometimes change default on new version that bites us

**Pull request merged**